torch/testing/_internal/opinfo/definitions/signal.py

```diff
@@ -0,0 +1,275 @@
+
+
+def error_inputs_window(op_info, device, *args, **kwargs):
+    # Tests for windows that have a negative size
+    yield ErrorInput(
+        SampleInput(-1, *args, dtype=torch.float32, device=device, **kwargs),
+        error_type=ValueError,
+        error_regex="requires non-negative window_length, got window_length=-1",
+    )
+
+    # Tests for window tensors that are not torch.strided, for instance, torch.sparse_coo.
+    yield ErrorInput(
+        SampleInput(3, *args, layout=torch.sparse_coo, device=device, dtype=torch.float32, **kwargs),
+        error_type=ValueError,
+        error_regex="is implemented for strided tensors only, got: torch.sparse_coo",
+    )
+
+    # Tests for window tensors that are not floating point dtypes, for instance, torch.long.
+    yield ErrorInput(
+        SampleInput(3, *args, dtype=torch.long, device=device, **kwargs),
+        error_type=ValueError,
+        error_regex="expects floating point dtypes, got: torch.int64",
+    )
+
+
+def error_inputs_exponential_window(op_info, device, **kwargs):
+    # Yield common error inputs
+    yield from error_inputs_window(op_info, device, 0.5, **kwargs)
+
+    # Tests for negative decay values.
+    yield ErrorInput(
+        SampleInput(3, tau=-1, dtype=torch.float32, device=device, **kwargs),
+        error_type=ValueError,
+        error_regex="Tau must be positive, got: -1 instead.",
+    )
+
+    # Tests for non-symmetric windows and a given center value.
+    yield ErrorInput(
+        SampleInput(3, center=1, sym=False, dtype=torch.float32, device=device),
+        error_type=ValueError,
+        error_regex="Center must be 'None' for non-symmetric windows",
+    )
+
+
+def error_inputs_gaussian_window(op_info, device, **kwargs):
+    # Yield common error inputs
+    yield from error_inputs_window(
+        op_info,
+        device,
+        0.5,  # std
+        **kwargs
+    )
+
+    # Tests for negative standard deviations
+    yield ErrorInput(
+        SampleInput(3, -1, dtype=torch.float32, device=device, **kwargs),
+        error_type=ValueError,
+        error_regex="Standard deviation must be positive, got: -1 instead.",
+    )
+
+
+def make_signal_windows_opinfo(
+        name, variant_test_name, ref, sample_inputs_func, error_inputs_func, *, skips=()
+):
+    r"""Helper function to create OpInfo objects related to different windows."""
+    return OpInfo(
+        name=name,
+        variant_test_name=variant_test_name,
+        ref=ref if TEST_SCIPY else None,
+        dtypes=floating_types_and(torch.bfloat16, torch.float16),
+        dtypesIfCUDA=floating_types_and(torch.bfloat16, torch.float16),
+        sample_inputs_func=sample_inputs_func,
+        error_inputs_func=error_inputs_func,
+        supports_out=False,
+        supports_autograd=False,
+        skips=skips,
+    )
+
+
+op_db: List[OpInfo] = [
+    make_signal_windows_opinfo(
+        name="signal.windows.cosine",
+        variant_test_name="signal.windows.cosine_default",
```

Comment:
I don't think we need the variant argument, as we just have one test per operation.